owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/SettingsFragmentTest.kt

```diff
@@ -0,0 +1,443 @@
+        }
+        Intents.init()
+    }
+
+    @After
+    fun tearDown() {
+        Intents.release()
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear().commit()
+    }
+
+    @Test
+    fun securityView() {
+        onView(withText(R.string.prefs_category_security)).check(matches(isDisplayed()))
+        assertEquals("security_category", categorySecurity.key)
+        assertEquals(context.getString(R.string.prefs_category_security), categorySecurity.title)
+        assertEquals(null, categorySecurity.summary)
+        assertTrue(categorySecurity.isVisible)
+
+        onView(withText(R.string.prefs_passcode)).check(matches(isDisplayed()))
+        assertEquals("set_pincode", prefPasscode.key)
+        assertEquals(context.getString(R.string.prefs_passcode), prefPasscode.title)
+        assertEquals(null, prefPasscode.summary)
+        assertTrue(prefPasscode.isVisible)
+        assertTrue(prefPasscode.isEnabled)
+        assertFalse(prefPasscode.isChecked)
+
+        onView(withText(R.string.prefs_pattern)).check(matches(isDisplayed()))
+        assertEquals("set_pattern", prefPattern.key)
+        assertEquals(context.getString(R.string.prefs_pattern), prefPattern.title)
+        assertEquals(null, prefPattern.summary)
+        assertTrue(prefPattern.isVisible)
+        assertTrue(prefPattern.isEnabled)
+        assertFalse(prefPattern.isChecked)
+
+        onView(withText(R.string.prefs_biometric)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_biometric_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_biometric)).check(matches(not(isEnabled())))
+        assertEquals("set_biometric", prefBiometric.key)
+        assertEquals(context.getString(R.string.prefs_biometric), prefBiometric.title)
+        assertEquals(context.getString(R.string.prefs_biometric_summary), prefBiometric.summary)
+        assertTrue(prefBiometric.isVisible)
+        assertFalse(prefBiometric.isEnabled)
+        assertFalse(prefBiometric.isChecked)
+
+        onView(withText(R.string.prefs_touches_with_other_visible_windows)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_touches_with_other_visible_windows_summary)).check(matches(isDisplayed()))
+        assertEquals("touches_with_other_visible_windows", prefTouchesWithOtherVisibleWindows.key)
+        assertEquals(
+            context.getString(R.string.prefs_touches_with_other_visible_windows),
+            prefTouchesWithOtherVisibleWindows.title
+        )
+        assertEquals(
+            context.getString(R.string.prefs_touches_with_other_visible_windows_summary),
+            prefTouchesWithOtherVisibleWindows.summary
+        )
+        assertTrue(prefTouchesWithOtherVisibleWindows.isVisible)
+        assertTrue(prefTouchesWithOtherVisibleWindows.isEnabled)
+        assertFalse(prefTouchesWithOtherVisibleWindows.isChecked)
+    }
+
+    @Test
+    fun passcodeOpen() {
```

Comment:
@abelgardep I thought we had some sort of naming convention for tests like the `ok - ko` thing or that test cases always start with the word `test` in front of them? However looking through the code I can see we do not have a consistent way of doing so? What is the right way to do it then?

Reply:
Yes, that's it. We need to have a consistent way of naming for tests. But the `ok - ko` [is not supported by the Android runtime](https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods) (at least for the moment). So, we will use it when possible: Unit tests. As you can see, we use the `ok - ko` in the `SettingsViewModelTest`